mct_quantizers/keras/quantize_wrapper.py

```diff
@@ -356,6 +356,10 @@ def call(self, inputs, training=None, **kwargs):
             Returns: tensors that simulate a quantized layer.
 
             """
+            if training is not None and not isinstance(training, bool):
+                raise TypeError("The 'training' argument must be either None or a boolean. "
```

Comment:
Use `Logger.critical` to raise exception with the proper error message.
Also, regarding the message, I would change to something like:

`f"Argument type mismatch: expected 'training' argument to be of type 'boolean' but got {type(training}}.`

Reply:
done